### PR TITLE
Difference HTML tables

### LINKED ISSUE #140: html difference table for model vs model/obs
### New feature type

New table and/or analytic calculation

### What is this new feature?

Create new HTML table for differences in values (means, etc.) for model vs model/obs and add to website. 

### Assistance required?

No, I will make a PR when the feature is ready

### Extra info

_No response_

### PR DESCRIPTION
**Pull request by justin-richling**
### Closes

#140

### Files Changed

```lib/adf_diag.py``` - add comparison table html file to website generation

```scripts/analysis/amwg_table.py``` - create new ```pandas dataframe``` for difference in mean values from case and baseline

### Tests

Ran with model vs model case, example comparison table found at: https://webext.cgd.ucar.edu/FWscHIST/f.c6_3_41d.FWscHIST.ne30_L58_macmic6_dribble.001/atm/ADF/html_table/amwg_table_comp.html

### CODE REVIEW

**Comment by justin-richling** [comment]
:warning: *** Hold off on merging until observations table is fixed *** :warning:

I need to configure html tables for observation cases; I can update when this PR is ready to go

**Comment by justin-richling** [comment]
@nusbaume This has been updated to ignore table comparison for comparing against observation since we aren't generating AMWG table for obs currently. 

When you're ready this PR is ready to be reviewed.

File: scripts/analysis/amwg_table.py [hunk 1]
```diff
@@ -297,5 +303,44 @@ def _write_html(f, out):
...
+    df_base = pd.read_csv(baseline)
+    df_comp = pd.DataFrame(dtype=object)
+
+    df_comp[['variable','unit','case']] = df_case[['variable','unit','mean']]
+    df_comp['baseline'] = df_base[['mean']]
+    
+    # ! * ! * ! * ! * ! * ! * ! * ! * ! * ! * ! * ! * ! * ! * ! * ! * ! * !
+    # Should this be an absolute value difference, or case minus baseline?
+    # ! * ! * ! * ! * ! * ! * ! * ! * ! * ! * ! * ! * ! * ! * ! * ! * ! * !
+    df_comp['diff'] = df_comp['baseline'].values-df_comp['case'].values
```
Comment (nusbaume) [inline scripts/analysis/amwg_table.py:326]: Should this difference be flipped?  That way positive numbers would mean that the test case has a positive bias, as opposed to meaning that the test case has a negative bias (which might require users to do a sign-flip in their head).

File: scripts/analysis/amwg_table.py [hunk 2]
```diff
@@ -297,5 +303,44 @@ def _write_html(f, out):
...
+    
+    html = df_comp.to_html(index=False, border=1, justify='center', float_format='{:,.3g}'.format)  # should return string
+    preamble = f"""<html><head></head><body><h1>AMWG Case Comparison<h1><h2>Test Case: {case_names[0]}<br/>Control Case: {case_names[1]}</h2>"""
+    ending = """</body></html>"""
+    with open(output_html_file_comp, 'w') as hfil:
+        hfil.write(preamble)
+        hfil.write(html)
+        hfil.write(ending)
+
+    #_write_html(output_csv_file_comp, output_html_file_comp)
```
Comment (nusbaume) [inline scripts/analysis/amwg_table.py:343]: Looks like old code here (might want to delete)?

File: lib/adf_diag.py [hunk 3]
```diff
@@ -1054,6 +1061,11 @@ def create_website(self):
                     #End if (table html file exists check)
                 #End for (case vs data)
 
+                #Check if comp table exists (if not, then obs are being compared and comp table is not created)
+                if comp_table_html_file:
+                    #Add comparison table to website dictionary
+                    amwg_tables["Case Comparison"] = comp_table.name
```
[the remark this answers is not in the document]
Comment (justin-richling) [reply lib/adf_diag.py:1067]: This was just for the time being. I can leave this and add a comment, then we can modify it when the time comes when we have multi-case, if that sounds good to you.

File: scripts/analysis/amwg_table.py [hunk 4]
```diff
@@ -297,5 +303,44 @@ def _write_html(f, out):
...
 
+def _df_comp_table(write_html,output_location,case_names):
+    import pandas as pd
+
+    # * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+    # first csv is case and second is baseline/obs 
+    # * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+    output_csv_file_comp = output_location / "amwg_table_comp.csv"
+    case = output_location/f"amwg_table_{case_names[0]}.csv"
+    baseline = output_location/f"amwg_table_{case_names[1]}.csv"
```
[the remark this answers is not in the document]
Comment (justin-richling) [reply scripts/analysis/amwg_table.py:314]: Just like in the ```lib/adf_diag.py``` comment above, I can leave this for single case for the time being and add a comment until we get to multi-case.

**Comment by justin-richling** [comment]
@nusbaume I've made some changes if you want to take another look and let me know if we need anything else here. 

Thanks!

**Review by nusbaume — APPROVED** [review]

**Pull request merged**